src/serena/config/serena_config.py

```diff
@@ -478,23 +462,24 @@ def from_config_file(cls, generate_if_missing: bool = True) -> "SerenaConfig":
             instance.projects.append(project)
 
         # set other configuration parameters
+        def get_value_or_default(field_name: str) -> Any:
+            return loaded_commented_yaml.get(field_name, get_dataclass_default(SerenaConfig, field_name))
+
         if is_running_in_docker():
             instance.gui_log_window_enabled = False  # not supported in Docker
         else:
-            instance.gui_log_window_enabled = loaded_commented_yaml.get("gui_log_window", False)
+            instance.gui_log_window_enabled = get_value_or_default("gui_log_window_enabled")
         instance.log_level = loaded_commented_yaml.get("log_level", loaded_commented_yaml.get("gui_log_level", logging.INFO))
-        instance.web_dashboard = loaded_commented_yaml.get("web_dashboard", True)
-        instance.web_dashboard_open_on_launch = loaded_commented_yaml.get("web_dashboard_open_on_launch", True)
-        instance.tool_timeout = loaded_commented_yaml.get("tool_timeout", DEFAULT_TOOL_TIMEOUT)
-        instance.trace_lsp_communication = loaded_commented_yaml.get("trace_lsp_communication", False)
-        instance.excluded_tools = loaded_commented_yaml.get("excluded_tools", [])
-        instance.included_optional_tools = loaded_commented_yaml.get("included_optional_tools", [])
-        instance.jetbrains = loaded_commented_yaml.get("jetbrains", False)
-        instance.token_count_estimator = loaded_commented_yaml.get(
-            "token_count_estimator", RegisteredTokenCountEstimator.TIKTOKEN_GPT4O.name
-        )
-        instance.default_max_tool_answer_chars = loaded_commented_yaml.get("default_max_tool_answer_chars", 150_000)
-        instance.ls_specific_settings = loaded_commented_yaml.get("ls_specific_settings", {})
+        instance.web_dashboard = get_value_or_default("web_dashboard")
```

Comment:
Just FYI @opcode81 - in the initial version of the change I was using get_value_or_default for the dashboard_host, but then pivoted to just not load it from the config. But the change is useful IMO, even if it doesn't relate to the issue at hand